### PR TITLE
feat: Adding support for external blobstore.

### PR DESCRIPTION
**Pull request by bikramnehra**
## Description
This PR intends to add support for on-demand external blobstore(AWS S3). Original PR: https://github.com/cloudfoundry-incubator/kubecf/pull/138.

fixes: https://github.com/cloudfoundry-incubator/kubecf/issues/563

## Motivation and Context

Uses the standard ops file provided from cf-deployment to provide this feature.

## How Has This Been Tested?

Locally on `minikube`. Following tests were performed:
- diego with singleton blobstore + smoke tests
- diego with external blobstore + smoke tests + CATs
- eirini with singleton blobstore + smoke tests
- eirini with external blobstore + smoke tests

See attached logs:
[test-results2.tar.gz](https://github.com/cloudfoundry-incubator/kubecf/files/4457457/test-results2.tar.gz)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code has security implications.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by loewenstein** [comment]
Is this a Duplo of https://github.com/cloudfoundry-incubator/kubecf/pull/138?

That one is only missing the image used via digest and instead has a fixed image in a separate registry.

**Comment by bikramnehra** [comment]
> Is this a duplicate of #138?
> 
> That one is only missing the image used via digest and instead has a fixed image in a separate registry.

yes @loewenstein this is based off https://github.com/cloudfoundry-incubator/kubecf/pull/138 
 with rebase onto `master` and updated `eirini-staging` images as the original branch was on a fork. Once we are done testing this branch we can merge directly onto `master` with link to original PR(see description).

**Comment by bikramnehra** [comment]
@viovanov let's not merge this yet. I am working on resolving merge conflicts. Also, this would need code review.

**Comment by f0rmiga** [comment]
@bikramnehra We want to preserve the commit history from the other PR. You will need to cherry-pick their commits and apply your fixes on top of it.

File: deploy/helm/kubecf/assets/operations/instance_groups/eirini.yaml [hunk 1]
```diff
@@ -148,9 +148,9 @@
           certs_secret_name: eirini-staging-secret
           cc_internal_api: https://cloud-controller-ng.service.cf.internal:9023
           eirini_address: {{ printf "https://eirini.%s:8484" .Release.Namespace }}
-          downloader_image: "eirini/recipe-downloader:0.3.0"
-          uploader_image: "eirini/recipe-uploader:0.3.0"
-          executor_image: "eirini/recipe-executor:0.3.0"
+          downloader_image: "eirini/recipe-downloader@sha256:80c213363762d3171fe401c331c480f0e8eb34fe781798eda95de1e7427b2771"
+          uploader_image: "eirini/recipe-uploader@sha256:cb70f76035866bf280e710247751f2f620c4c6e6e6950a6e730bde01535bebf1"
+          executor_image: "eirini/recipe-executor@sha256:c072f828b86766778d027041e394eaad9cfb55c3b5f652a50be8455d1dad3bf4"
```
Comment (mook-as) [inline deploy/helm/kubecf/assets/operations/instance_groups/eirini.yaml:153]: We should have a follow-up task to make this customizable, like all the other releases are.

File: deploy/helm/kubecf/templates/s3-blobstore.yaml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+apiVersion: "v1"
+kind: "Secret"
+type: Opaque
+metadata:
+  name: var-blobstore-access-key-id
+  namespace: {{ .Release.Namespace | quote }}
+  labels:
+    app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/name: {{ default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" | quote }}
```
Comment (mook-as) [inline deploy/helm/kubecf/templates/s3-blobstore.yaml:12]: Shouldn't this use the [`kubecf.name`](https://github.com/cloudfoundry-incubator/kubecf/blob/d393c89bcf0504f6b9f2509ef59fb31ced692925/deploy/helm/kubecf/templates/_helpers.tpl#L5) macro?

(I'm surprised we don't have a macro for standard labels.)

File: deploy/helm/kubecf/assets/operations/instance_groups/eirini.yaml (hunk 1 above)
Comment (bikramnehra) [reply deploy/helm/kubecf/assets/operations/instance_groups/eirini.yaml:153]: created: https://github.com/cloudfoundry-incubator/kubecf/issues/640

File: deploy/helm/kubecf/templates/s3-blobstore.yaml (hunk 2 above)
Comment (bikramnehra) [reply deploy/helm/kubecf/templates/s3-blobstore.yaml]: DONE. nice catch.

**Comment by f0rmiga** [comment]
@mook-as I'm also bummed. @kramerul @phil9909 Do you guys want us to work more on the commit history to keep the original commits and apply the fixes on top, or are you fine with the co-authored commit that @bikramnehra created (https://github.com/cloudfoundry-incubator/kubecf/pull/608/commits/7ae73b1210ff249b5e7b1ff0ea9a53709dd215d0)?

**Comment by wonderix** [comment]
I'm fine with everything you want to do.

**Comment by kramerul** [comment]
Sorry, wrong user. I'm fine with everything you want to do.

**Comment by phil9909** [comment]
No problem for me

**Pull request merged**